zerver/views/invite.py

```diff
@@ -103,7 +103,7 @@ def revoke_user_invite(
     except PreregistrationUser.DoesNotExist:
         raise JsonableError(_("No such invitation"))
 
-    if prereg_user.referred_by.realm != user_profile.realm:
+    if prereg_user.referred_by is not None and prereg_user.referred_by.realm != user_profile.realm:
```

Comment:
I think the original code is subtly wrong, and thus change would make it more so.  We should be just checking `prereg_user.realm != user_profile.realm`.  Can you do this in a separate commit?